app/services/carry_over_application.rb

```diff
@@ -26,10 +26,6 @@ def application_from_current_cycle?
   end
 
   def recruitment_cycle_year
-    if EndOfCycleTimetable.between_cycles_apply_2? && EndOfCycleTimetable.current_cycle?(@application_form)
-      RecruitmentCycle.next_year
-    else
-      RecruitmentCycle.current_year
-    end
+    application_form.recruitment_cycle_year + 1
```

Comment:
This makes the `raise_if_application_from_current_cycle` and `application_from_current_cycle?` methods above a bit redundant, they are basically testing `n + 1 == n` now. Not sure whether we need the check now that you've covered the issue of double carry overs in https://github.com/DFE-Digital/apply-for-teacher-training/pull/3063/files. So maybe just remove those methods?